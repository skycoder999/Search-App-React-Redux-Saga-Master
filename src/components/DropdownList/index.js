import React from "react";
import PropTypes from "prop-types";
import DropdownItem from "../DropdownItem";
import loader from "../../assets/images/loader.gif";

const DropdownList = ({ items, query, selected, loading, visible, onSelect }) => {
  return (
    <>
      {!selected && query ? (
        <div className={"suggest-list" + (visible ? " visible" : "")}>
          {!loading && (!items || !items.length) ? <p className="suggest-list__msg">Oops! No results found.</p> : null}
          {!loading && items && items.length ? items.map(item => <DropdownItem item={item} onSelect={onSelect} key={item.id} />) : null}
          {loading ? (
            <p className="suggest-list__msg">
              <img src={loader} alt="" /> Loading results...
            </p>
          ) : null}
        </div>
      ) : null}
    </>
  );
};

DropdownList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  loading: PropTypes.bool,
  onSelect: PropTypes.func,
  query: PropTypes.string,
  selected: PropTypes.string,
  visible: PropTypes.bool
};

export default React.memo(DropdownList);
