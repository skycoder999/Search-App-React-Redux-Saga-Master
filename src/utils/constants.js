export const API_SEARCH_PATH = "https://api.themoviedb.org/3/search/movie?api_key=f9dba24a3b8ed9425600eb5d5fbd9a93";
export const API_POSTER_PATH = "https://image.tmdb.org/t/p/w92";
export const API_POSTER_PATH_BIG = "https://image.tmdb.org/t/p/w500";
export const API_THROTTLE_DELAY = 300;
export const API_RESULTS_LIMIT = 10;
export const TEXT_SEARCH_PLACEHOLDER = "Type here to search movies...";
export const TEXT_LOADING = "Loading results...";
export const TEXT_NO_RESULTS = "Oops! No results found.";

