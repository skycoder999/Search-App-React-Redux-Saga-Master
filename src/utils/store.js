import { createStore, applyMiddleware } from "redux";
import { logger } from "redux-logger";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../reducers/root.reducer";
import rootSaga from "../sagas/root.saga";

const configureStore = (initialState = {}) => {
  const DEBUG = false;
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [sagaMiddleware, DEBUG && logger];
  const store = createStore(rootReducer, initialState, applyMiddleware(...middlewares.filter(Boolean)));
  sagaMiddleware.run(rootSaga);
  return store;
};

export default configureStore;
