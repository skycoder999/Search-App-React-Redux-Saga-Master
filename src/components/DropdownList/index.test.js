import DropdownList from "./index";
import { TEXT_LOADING } from "../../utils/constants";

let wrapper, consoleError;
let items = [
  {
    id: 1,
    original_title: "Sample Title",
    overview: "Sample Overview",
    poster_path: "some_poster.jpg",
    vote_average: 4,
    release_date: "2019-02-01"
  }
];

beforeEach(() => {
  wrapper = shallow(<DropdownList items={items} query="star" />);
  consoleError = console.error;
  console.error = jest.spyOn(console, "error");
});

afterEach(() => {
  console.error = consoleError;
});

describe("DropdownList Component", () => {
  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
