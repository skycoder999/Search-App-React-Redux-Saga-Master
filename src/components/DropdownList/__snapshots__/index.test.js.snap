// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownList Component matches the snapshot 1`] = `ShallowWrapper {}`;
